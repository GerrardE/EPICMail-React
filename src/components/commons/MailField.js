import React from 'react';
import PropTypes from 'prop-types';
import { Button } from './Button';

export const MailField = ({
  id, classType, email, createdon, title, body, onClick
}) => (
    <div key={id} className={classType}>
      <div className="top">
        <span className="title">{email}</span>
        <span className="time-right">{createdon}</span>
      </div>
      <p className="lead">{title}</p>
      <p className="para">{body}</p>
      {/* <Button
        linkClass="viewbtn"
        btnName="VIEW"
        value={id}
        onClick={onClick}
      />
      <Button
        linkClass="deletebtn"
        btnName="DELETE"
      /> */}
    </div>
  );

MailField.propTypes = {
  id: PropTypes.number,
  classType: PropTypes.string,
  email: PropTypes.string,
  createdon: PropTypes.string,
  title: PropTypes.string,
  body: PropTypes.string,
  btnName: PropTypes.string,
};

MailField.defaultProps = {
  classType: "container-chat darker-new",
  email: "EPICMail Team",
};

export default MailField;
