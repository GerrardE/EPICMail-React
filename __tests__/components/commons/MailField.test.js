import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MailField from '../../../src/components/commons/MailField';

const props = {
  id: 1, 
  classType: '', 
  email: '',
  createdon: '', 
  title: '', 
  body: '', 
  onClick: jest.fn()
}

const setup = () => {
  const wrapper = shallow(<MailField {...props} />);
  return wrapper
}

describe('<MailField />', () => {
  test('should render thecMailField component properly', () => {
    let wrapper;
    wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
