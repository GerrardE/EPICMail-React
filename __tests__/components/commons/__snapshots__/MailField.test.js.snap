// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MailField /> should render thecMailField component properly 1`] = `
<div
  className=""
  key="1"
>
  <div
    className="top"
  >
    <span
      className="title"
    />
    <span
      className="time-right"
    />
  </div>
  <p
    className="lead"
  />
  <p
    className="para"
  />
</div>
`;
